src/math/trig.c

```diff
@@ -94,9 +208,192 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
 		xtmp = *xn >> (b_idx + 1);
 		ytmp = *b_yn >> (b_idx + 1);
 	}
-	/* Q2.30 format */
+	/* Q2.30 format -sine, cosine*/
 	*th_cdc_fxp = th_rad_fxp;
+}
 
+/**
+ * \CORDIC-based approximation for inverse cosine
+ * Arguments	: int32_t cosvalue
+ *		  int16_t numiters
+ *		  const int32_t LUT[CORDIC_30B_ITABLE_SIZE]
+ * Return Type	: int32_t
+ */
+int32_t is_scalar_cordic_acos(int32_t cosvalue, int16_t numiters,
+			      const int32_t *LUT)
+{
+	int32_t xdshift;
+	int32_t ydshift;
+	int32_t yshift;
+	int32_t xshift;
+	int32_t x = 0;
+	int32_t y = 0;
+	int32_t z = 0;
+	int32_t sign;
+	int32_t b_i;
+	int32_t i;
+	int16_t j;
+	int16_t k;
```

Comment:
I wonder if code speed could improve with i, j, k as  plain "int". numiters is 16 bit, then it's copied to i that is 32 bit, then the type cast in loop for k would not be needed. I'm not sure about it, but int should give more freedom for compiler  to choose the type for integers those contain small value numbers. 16 bit is a slow to compute type in xtensa. But it could be that there's no impact.